app/experimenter/nimbus-ui/src/components/FormMetrics/index.tsx

```diff
@@ -0,0 +1,196 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import Form from "react-bootstrap/Form";
+import Alert from "react-bootstrap/Alert";
+import { getExperiment } from "../../types/getExperiment";
+import { getConfig_nimbusConfig_probeSets } from "../../types/getConfig";
+import { useExitWarning } from "../../hooks";
+import Select, { ActionMeta, ValueType } from "react-select";
+
+type FormMetricsProps = {
+  experiment: getExperiment["experimentBySlug"];
+  probeSets: (getConfig_nimbusConfig_probeSets | null)[] | null;
+  isLoading: boolean;
+  isServerValid: boolean;
+  submitErrors: Record<string, string[]>;
+  onSave: (data: Record<string, any>, reset: Function) => void;
+  onNext?: (ev: React.FormEvent) => void;
+};
+
+const FormMetrics = ({
+  experiment,
+  probeSets,
+  isLoading,
+  isServerValid,
+  submitErrors,
+  onSave,
+  onNext,
+}: FormMetricsProps) => {
+  const { handleSubmit, reset, formState } = useForm({
+    mode: "onTouched",
+  });
+  const { isSubmitted, isDirty } = formState;
+  const [selectedPrimaryProbeSetIds, setSelectedPrimaryProbeSetIds] = useState<
+    string[]
+  >([]);
+  const [selectedSecondaryProbeSets, setSelectedSecondaryProbeSets] = useState<
+    string[]
+  >([]);
+
+  const isValid = isServerValid && formState.isValid;
+  const isDirtyUnsaved = isDirty && !(isValid && isSubmitted);
+
+  const shouldWarnOnExit = useExitWarning();
+  useEffect(() => {
+    shouldWarnOnExit(isDirtyUnsaved);
+  }, [shouldWarnOnExit, isDirtyUnsaved]);
+
+  const handleSubmitAfterValidation = useCallback(
+    (data: Record<string, any>) => {
+      if (isLoading) return;
+      onSave(data, reset);
+    },
+    [isLoading, onSave, reset],
+  );
+
+  const handleNext = useCallback(
+    (ev: React.FormEvent) => {
+      ev.preventDefault();
+      onNext!(ev);
+    },
+    [onNext],
+  );
+
+  const handlePrimaryProbeSetsChange = (
+    selectedOptions: { label: string; value: string }[] | null,
+  ) => {
+    const probeSetIds = selectedOptions?.map((option) => option?.value) || [];
+    setSelectedPrimaryProbeSetIds(probeSetIds);
+  };
+
+  const handleSecondaryProbeSetsChange = (
+    selectedOptions: { label: string; value: string }[] | null,
+  ) => {
+    const probeSetIds = selectedOptions?.map((option) => option?.value) || [];
+    setSelectedSecondaryProbeSets(probeSetIds);
+  };
+
+  const primaryProbeSetOptions: { label: string; value: string }[] = [];
+  const secondaryProbeSetOptions: { label: string; value: string }[] = [];
+
+  if (probeSets) {
+    for (const probeSet of probeSets) {
+      if (probeSet && selectedSecondaryProbeSets.indexOf(probeSet.id) === -1) {
+        primaryProbeSetOptions.push({
+          label: probeSet.name,
+          value: probeSet.id,
+        });
+      }
+      if (probeSet && selectedPrimaryProbeSetIds.indexOf(probeSet.id) === -1) {
```

Comment:
These two checks could use `Array.includes`, e.g. `selectedPrimaryProbeSetIds.includes(probeSet.id)` (and you could throw `if (!probeSet) continue;` at the top of the for-loop)